### PR TITLE
Add custom HTTP headers to fetchers to allow resource time-based negotiation

### PR DESCRIPTION
**Pull request by julianrojas87**

### CODE REVIEW

File: lib/BrowserHttpFetcher.js
Comment (pietercolpaert) [inline lib/BrowserHttpFetcher.js]: This will break backwards compatibility. Can we add the customHeaders as the last parameter? We should then check whether maxParallel is an integer or not, and if it appears not to be an integer, it will be the customHeaders parameter. This  way, we have a good backwards compatible strategy I guess...

File: lib/NodeHttpFetcher.js
Comment (pietercolpaert) [inline lib/NodeHttpFetcher.js]: Same comment here: we want backwards compatibility, so I’d add it at the end

**Review by pietercolpaert — CHANGES_REQUESTED** [review]
Thanks! 2 minor comments

**Comment by julianrojas87** [comment]
Changes are impemented.

**Comment by pietercolpaert** [comment]
Blocked until backwards compatibility is tested

File: README.md
Comment (pietercolpaert) [inline README.md]: Why key and value, and why not just `{"headers":{"accept-datatime":"2017-03-11T17:00:00.000Z"}}`?

File: lib/BrowserHttpFetcher.js
Comment (pietercolpaert) [inline lib/BrowserHttpFetcher.js:12]: we should also have maxParallel en enableCache as properties to have the same interface outside of the browser

File: README.md
Comment (pietercolpaert) [inline README.md:38]: Good idea to put default headers in ldfetch itself!

File: README.md
Comment (pietercolpaert) [inline README.md:40]: yet we should also be able to support  `fetch.get(url, options)`

File: lib/NodeHttpFetcher.js
Comment (pietercolpaert) [inline lib/NodeHttpFetcher.js:18]: what’s happening here? Why do you always enable cache?

**Review by pietercolpaert — CHANGES_REQUESTED** [review]
Before merge, I’d still like to see these comments addressed

**Comment by pietercolpaert** [comment]
Thanks!

**Pull request merged**
